### PR TITLE
[WIP] rx.color

### PR DESCRIPTION
**Pull request by Alek99**
Initial thoughts on what it would look like 

```python
"""Welcome to Reflex! This file outlines the steps to create a basic app."""
"""Welcome to Reflex! This file outlines the steps to create a basic app."""
from rxconfig import config

import reflex as rx
import reflex.components.radix.themes as rdxt
import reflex.components.radix.primitives as rdxp
from reflex.utils.serializers import get_serializer
import random

class State(rx.State):
    pass

def index() -> rx.Component:
    return rdxt.theme(
        rx.text(
            "test",
            bg=rx.color("red"),
        ),
    )     
    
# Add state and page to the app.
app = rx.App(
    theme=rdxt.theme(
        appearance="dark", has_background=True, radius="none", accent_color="gold" 
    ), 
) 
 
app.add_page(index) 
  ```

### CODE REVIEW

File: reflex/constants/colors.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    Args:
+        color: The color to serialize.
+
+    Returns:
+        The serialized color.
+    """
+    print(f"--var({color.color}-{color.shade})")
+    if color.alpha:
+        return f"var(--{color.color}-{color.shade}-alpha)"
```
Comment (masenf) [inline reflex/constants/colors.py:70]:
```suggestion
        return f"var(--{color.color}-a{color.shade})"
```

i thought the alphas used an `a` prefix on the shade

**Review by masenf — COMMENTED** [review]
dude, i _dig_ this approach! awesome! 🤩

File: reflex/constants/colors.py (hunk 1 above)
Comment (Alek99) [reply reflex/constants/colors.py:70]: ah your correct

**Review by picklelo — COMMENTED** [review]
Looks good!

* I think this should be accessible as `rx.colors` rather than as a separate import
* Can we add some tests in test_serializers.py

File: reflex/constants/colors.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        "amber",
+        "gold",
+        "bronze",
+        "gray",
+    ]
+
+ShadeType= Literal[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+
+@dataclass
+class Colors:
```
Comment (picklelo) [inline reflex/constants/colors.py:46]: Let's name this class `Color`, then call it as `rx.color(...)`

**Comment by Alek99** [comment]
If the app isn't in a radix theme the colors won't work as they ref radix color palette. I don't see an easy way to check this though to throw an error if this happens.

**Comment by masenf** [comment]
> If the app isn't in a radix theme

i think the plan is still to always apply a default theme, so this might be a moot point. either way you should be able to get `get_app().theme` and see if it's set to an instance of `rdxt.Theme`

**Review by picklelo — APPROVED** [review]

**Pull request merged**